src/OpenTelemetry.Instrumentation.AspNetCore/MeterProviderBuilderExtensions.cs

```diff
@@ -34,7 +38,21 @@ public static class MeterProviderBuilderExtensions
     /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
     public static MeterProviderBuilder AddAspNetCoreInstrumentation(
         this MeterProviderBuilder builder)
-        => AddAspNetCoreInstrumentation(builder, name: null, configureAspNetCoreInstrumentationOptions: null);
+    {
+        Guard.ThrowIfNull(builder);
+
+#if NET8_0_OR_GREATER
+        return builder
```

Comment:
not a blocker for this PR, but we need to [measure perf before stable release.](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4799#issuecomment-1743681789)

If the perf overhead is significant, we may want to only add those meter which would give the metric we want (based on otel spec), and ask users to opt-in to more based on needs.
If the overhead is non-significant, then a doc about view recipe to disable would be sufficient.